src/Security/Authentication/Authenticator.php

```diff
@@ -94,4 +93,14 @@ protected function hasRequiredFields(array $array): bool
 
         return true;
     }
+
+    protected function getRequest(Credentials $credentials): Request
+    {
+        return new Request(
```

Comment:
just an idea: can we create also `Authenticate` request class and pass the `Credentials` instance into the constructor? so all the requests would be stored in one unified manner